.github/workflows/docker.yml

```diff
@@ -0,0 +1,48 @@
+name: Docker Image CI
+
+on: push
+  
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+   
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ghcr.io/${{ github.repository }}
+          # generate Docker tags based on the following events/attributes
+          tags: |
+            # set latest tag for main branch
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=ref,event=branch
+            type=sha
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+   
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2    
+
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This would need a token with write permission.

Reply:
The token mentioned in this context is automatically generated. You can find more information about this in the following link: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow. However, in order to utilize write permissions in your repository actions, you need to enable them in your action's settings. Please refer to the attached screenshot for guidance.
<img width="841" alt="Captura de pantalla 2023-03-18 a las 11 14 19" src="https://user-images.githubusercontent.com/1876752/226101956-5211349a-1262-4684-a40a-e174f14d5dea.png">